processors/base/helpers/__init__.py

```diff
@@ -266,3 +277,56 @@ def safe_prepend(prepend_string, string):
         string = '%s%s' % (prepend_string, string)
 
     return string
+
+
+def get_canonical_location_name(location):
+    """Find the canonical location name according to the
+    passed entry
+
+    Args:
+        location (str): the location to be normalized
+    """
+
+    # This auxiliar file is provided by the dataset repository of github.com and
+    # can be checked on its latest version on the project website
+    # See: https://github.com/datasets/country-codes/blob/master/data/country-codes.csv
+    #
+    # To create e new entry (or update an existing one), add a new line (or find the
+    # existing one) and fill in the informations according to the header (line 1) order,
+    # separating fields using comma
+    CSV_PATH = 'data/countries.csv'
+    ALPHA_3_HEADER = 'ISO3166-1-Alpha-3'
+    DISTANCE_SCORER = fuzz.token_sort_ratio
+    SCORE_INDEX = 1
+
+    clean_string = lambda u: re.sub(u"[^\w\d'\s]+", '', u).lower()
+    # Try to fetch canonical name directly from the iso-3166 country standard
+    try:
+        return iso3166.countries.get(clean_string(location)).name
+    # If country isn't found on iso-3166 country list
+    except KeyError:
+        with open(os.path.join(os.path.dirname(__file__), CSV_PATH), 'r') as csv_file:
+            reader = csv.DictReader(csv_file)
+            # Store information about the current match
+            current_match = location
+            current_score = float("-inf")
+            for country in reader:
+                # Extract relevant comparative info from the current country
+                relevant_info = [unicode(country[field], encoding="utf-8")
+                                 for field in reader.fieldnames[0:5]]
+                # Do some string clean up and extract the most similar field (according
+                # to Levenshtein scoring) over all the location relevant informations
+                cleaned_location = clean_string(location)
+                location_info = [location_info.lower() for location_info in relevant_info]
+                choice = process.extractOne(cleaned_location, location_info, scorer=DISTANCE_SCORER)
```

Comment:
Resolved. The reason why I hadn't done that yet is because since I'm not using the `match` variable (I'm fetching the current match directly from the iso3166 lib using the line information) I thought that would be a bad practice